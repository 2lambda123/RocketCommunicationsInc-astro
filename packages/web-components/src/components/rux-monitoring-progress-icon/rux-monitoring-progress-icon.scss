:host {
    display: inline-flex;
    padding: 0;
}

:host([hidden]) {
    display: none;
}

*,
*::before,
*::after {
    box-sizing: border-box;
}

.rux-advanced-status {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    position: relative;
    margin: 0;
    line-height: 0;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
}

.rux-advanced-status__icon-group {
    display: flex;
    position: relative;
    margin: var(--spacing-0) auto;
    width: calc(var(--spacing-16) + var(--spacing-4)); //5rem
    padding-top: var(--spacing-1);
}

rux-status {
    position: absolute;
    top: 0;
    margin: var(--spacing-050) var(--spacing-2);
}

.rux-advanced-status__badge {
    display: block;
    z-index: 2;
    order: 3;
    position: absolute;
    right: 0;
    top: 0;
    box-shadow: inset 0 0 0 1px var(--color-text-secondary);
    border-radius: var(--radius-base);
    padding: var(--spacing-2) var(--spacing-1); //8px 4px
    color: var(--color-palette-neutral-000);
    font-family: var(--font-body-3-font-family);
    font-size: var(--font-body-3-font-size);
    font-weight: var(--font-body-3-font-weight);
    letter-spacing: var(--font-body-3-letter-spacing);
    text-align: center;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    background-color: #000;
}

.rux-advanced-status__label {
    color: var(--color-text-primary);
    font-family: var(--font-body-2-font-family);
    font-size: var(--font-body-2-font-size);
    font-weight: var(--font-body-2-font-weight);
    letter-spacing: var(--font-body-2-letter-spacing);
    line-height: var(--font-body-2-line-height);
    text-align: center;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
    width: 100%;
    max-width: 6.25rem;
}

.rux-advanced-status__sublabel {
    font-family: var(--font-body-3-font-family);
    font-size: var(--font-body-3-font-size);
    font-weight: var(--font-body-3-font-weight);
    letter-spacing: var(--font-body-3-letter-spacing);
    line-height: var(--font-body-3-line-height);
    color: var(--color-text-primary);
    display: block;
}

.rux-advanced-status__icon-group rux-status::part(status) {
    line-height: 0;
}

.rux-advanced-status__hidden {
    display: none;
}

svg {
    margin: 0 auto;
    width: 2.5rem; //40px
    height: 2.5rem; //40px
}
svg.rux-status--off {
    stroke: var(--status-symbol-color-fill-off-on-dark, rgb(158, 167, 173));
    fill: var(--status-symbol-color-fill-off-on-dark, rgb(158, 167, 173));
}

svg.rux-status--standby {
    stroke: var(--status-symbol-color-fill-standby-on-dark, rgb(45, 204, 255));
    fill: var(--status-symbol-color-fill-standby-on-dark, rgb(45, 204, 255));
}

svg.rux-status--normal {
    stroke: var(--status-symbol-color-fill-normal-on-dark, rgb(86, 240, 0));
    fill: var(--status-symbol-color-fill-normal-on-dark, rgb(86, 240, 0));
}

svg.rux-status--caution {
    stroke: var(--status-symbol-color-fill-caution-on-dark, rgb(252, 232, 58));
    fill: var(--status-symbol-color-fill-caution-on-dark, rgb(252, 232, 58));
}

svg.rux-status--serious {
    stroke: var(--status-symbol-color-fill-serious-on-dark, rgb(255, 179, 0));
    fill: var(--status-symbol-color-fill-serious-on-dark, rgb(255, 179, 0));
}

svg.rux-status--critical {
    stroke: var(--status-symbol-color-fill-critical-on-dark, rgb(255, 56, 56));
    fill: var(--status-symbol-color-fill-critical-on-dark, rgb(255, 56, 56));
}

[data-progress] svg {
    transition: stroke-dashoffset 0.367s, stroke 0.367s;
    transform-origin: 50% 50%;
}

.rux-advanced-status__progress {
    color: var(--color-text-primary);
    font-family: var(--font-body-3-font-family);
    font-size: var(--font-body-3-font-size);
    font-weight: var(--font-body-3-font-weight);
    letter-spacing: var(--font-body-3-letter-spacing);
    line-height: var(--font-body-3-line-height);
    position: absolute;
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: calc(
        100% - var(--spacing-1)
    ); // accounts for the padding-top on the icon group
    text-align: center;
}
