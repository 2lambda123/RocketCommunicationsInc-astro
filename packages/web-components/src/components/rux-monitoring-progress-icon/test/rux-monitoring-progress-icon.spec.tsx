import { newSpecPage } from '@stencil/core/testing'
import { RuxMonitoringProgressIcon } from '../rux-monitoring-progress-icon'

describe('rux-monitoring-progress-icon', () => {
    it('renders', async () => {
        const page = await newSpecPage({
            components: [RuxMonitoringProgressIcon],
            html: `
      <rux-monitoring-progress-icon
        progress=70
        max="100"
        label="Label"
        sublabel="sublabel"
        notifications="345678"
      ></rux-monitoring-progress-icon>`,
        })
        expect(page.root).toMatchSnapshot()
    })

    it('applies custom range arrays properly', () => {
        const progressIcon = new RuxMonitoringProgressIcon()
        progressIcon.min = 100
        progressIcon.max = 1100
        progressIcon.progress = 675
        progressIcon.range = [
            {
                threshold: 300,
                status: 'standby',
            },
            {
                threshold: 775,
                status: 'normal',
            },
            {
                threshold: 1100,
                status: 'serious',
            },
        ]
        progressIcon._updateProgress()
        expect(progressIcon.status).toBe('normal')
        expect(
            Math.ceil(
                ((progressIcon.progress - progressIcon.min) /
                    (progressIcon.max - progressIcon.min)) *
                    100
            )
        ).toBe(58)

        progressIcon.progress = 1100
        progressIcon._updateProgress()
        expect(progressIcon.status).toBe('serious')
    })

    it('works with empty range array', () => {
        const progressIcon = new RuxMonitoringProgressIcon()

        progressIcon.progress = 50
        progressIcon.range = []
        progressIcon.componentWillLoad()
        expect(progressIcon.status).toBe('caution')
    })
    it('changes progress value to equal max if passed in progress is greater', () => {
        const progressIcon = new RuxMonitoringProgressIcon()
        progressIcon.progress = 101
        progressIcon.componentWillLoad()
        expect(progressIcon.progress).toEqual(progressIcon.max)
    })
    it('changes progress value to equal min if passed in progress is less', () => {
        const progressIcon = new RuxMonitoringProgressIcon()
        progressIcon.progress = -10
        progressIcon.componentWillLoad()
        expect(progressIcon.progress).toEqual(progressIcon.min)
    })
})
