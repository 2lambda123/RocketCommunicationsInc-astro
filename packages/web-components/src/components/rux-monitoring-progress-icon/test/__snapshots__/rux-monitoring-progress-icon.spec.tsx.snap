// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rux-monitoring-progress-icon renders 1`] = `
<rux-monitoring-progress-icon label="Label" max="100" notifications="345678" progress="70" sublabel="sublabel">
  <mock:shadow-root>
    <div class="rux-advanced-status" id="rux-advanced-status__icon" part="container" title="345678 Label sublabel">
      <div class="rux-advanced-status__icon-group" part="icon-group">
        <rux-status part="status-icon" status="serious"></rux-status>
        <svg class="rux-status--serious" part="radial-progress" viewBox="0 0 128 128" xmlns="http://www.w3.org/2000/svg">
          <g id="progress">
            <circle cx="60" cy="60" fill="transparent" r="56" stroke="rgba(40, 63, 88, 1)" stroke-width="10" transform="rotate(-90 61 60)"></circle>
            <circle class="progress-ring__circle" cx="60" cy="60" fill="transparent" r="56" stroke-dasharray="351.8583772 351.8583772" stroke-dashoffset="105.55751316061708" stroke-linecap="round" stroke-width="10" transform="rotate(-90 61 60)"></circle>
          </g>
        </svg>
        <div class="rux-advanced-status__progress" part="progress-display">
          70%
        </div>
        <div class="rux-advanced-status__badge" part="monitoring-badge">
          345K
        </div>
      </div>
      <div class="rux-advanced-status__label" part="monitoring-label">
        <span>
          Label
        </span>
        <span class="rux-advanced-status__sublabel" part="monitoring-sublabel">
          sublabel
        </span>
      </div>
    </div>
  </mock:shadow-root>
</rux-monitoring-progress-icon>
`;
